Automattic/wp-calypso
# Marketplace: Add notice to indicate plugins not tested with recent WordPress version

This PR adds a notice to the plugin details indicating if the plugin was tested more than 3 versions of WordPress above. Thus the user will identify that the plugin might be obsolete

Solves #https://github.com/Automattic/dotcom-forge/issues/3312

|Before | After|
|-------|------|
| ![CleanShot 2023-08-28 at 14 47 59@2x](https://github.com/Automattic/wp-calypso/assets/3519124/a76c9e55-7275-4b9b-868c-b28f454be339) | ![CleanShot 2023-08-28 at 17 39 09@2x](https://github.com/Automattic/wp-calypso/assets/3519124/ec0bbb32-1281-4734-aadf-6779363f6543))  |

## Proposed Changes

* Adds a query to WordPress.org to retrieve the versions. The API url used is https://api.wordpress.org/core/version-check/1.7/
* Adds a check that compares the versions to the previous 3 versions. 
* Displays a warning message if the version is less than 3 versions of the latest WP version. 

For example, if the list of WP versions are: `[6.3", "6.2.2", "6.1.3", "6.0.5", "5.9.7", "5.8.7"]`. The warning will be displayed if the plugin has been tested up to `"5.9.7"` or below

## Testing Instructions



* Apply the patch
* Navigate to a deprecated plugin, e.g. `/plugins/restrict-media-library-access`
* Make sure you can see the notice 
* Navigate to a maintained plugin, e.g. `/plugins/woocommerce`
* Make sure you do not see any additional notice

### Additional Things to check
- The wording
- The `search` hyperlink in the message

## Pre-merge Checklist
